spectator/writer/line_buffer.py

```diff
@@ -0,0 +1,25 @@
+from typing import List
+
+
+class LineBuffer:
+    """Line buffer for Writers."""
+
+    def __init__(self, max_size: int) -> None:
+        self._size: int = 0
+        self._max_size: int = max_size
+        self._lines: List[str] = []
+
+    def __len__(self) -> int:
+        return self._size
+
+    def append(self, line: str) -> bool:
+        """Return value indicates whether the buffer should be flushed."""
+        self._lines.append(line)
+        self._size += len(line)
+        return self._size > self._max_size
+
+    def flush(self) -> str:
+        lines = "\n".join(self._lines)
```

Comment:
Should the line feeds be considered when computing the size of the buffer?

Reply:
Given that this is a dynamic language, it's ok to be a bit over the max size of the buffer, and it keeps the implementation simple. It's mostly used as an indication that that buffer should be flushed before accumulating more.

The append to the buffer that pushes it over the `max_size` can be quite a bit larger than the few extra new line characters that are used to construct the final line sent to spectatord.

Mostly curious to hear an opinion on how strict we should be in terms of adhering to the max buffer size.